compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -254,6 +291,20 @@ static LogicalResult setContractConfig(func::FuncOp entryPoint,
   if (!linalg::isaContractionOpInterface(op) || op.getNumParallelLoops() < 2) {
     return failure();
   }
+
+  // Also exclude the case of matvec, which has only one non-unit parallel dim.
+  // They should go down different pipelines.
+  int nonUnitParallelDimCount = 0;
```

Comment:
Interesting... I understand that it probably needs a different pipeline.... but I dont understand why a `linalg.matmul` and `linalg.batch_matmul` of unit dimension can go down this pipeline. Can you give some more context here.

Reply:
If I understand your question correctly, `linalg.matmul` and `linalg.batch_matmul` even of unit dimension would go down this pipeline as it satisfies the condition to go through this pipeline which is `linalg::isaContractionOpInterface && op.getNumParallelLoops() <  2`. 

lmk if that's not really answering the question :)

Reply:
A little more context, we're doing the same thing on the SPIR-V side because we don't have any better pipeline for handling matvecs (and this is just checking for matmuls that happen to be a matvec).

Reply:
+1. It's trying to avoid using matmul pipelines for what's really a matvec, to improve performance.